src/components/ha-hls-player.ts

```diff
@@ -93,34 +94,40 @@ class HaHLSPlayer extends LitElement {
   }
 
   private async _getUseExoPlayer(): Promise<boolean> {
-    return false;
+    if (!this.hass!.auth.external || !this.allowExoPlayer) {
+      return false;
+    }
+    const externalConfig = await getExternalConfig(this.hass!.auth.external);
+    return externalConfig && externalConfig.hasExoPlayer;
   }
 
   private async _startHls(): Promise<void> {
-    let hls: any;
     const videoEl = this._videoEl;
-    this._useExoPlayer = await this._getUseExoPlayer();
-    if (!this._useExoPlayer) {
-      hls = ((await import(/* webpackChunkName: "hls.js" */ "hls.js")) as any)
-        .default as HLSModule;
-      let hlsSupported = hls.isSupported();
-
-      if (!hlsSupported) {
-        hlsSupported =
-          videoEl.canPlayType("application/vnd.apple.mpegurl") !== "";
-      }
+    const url = this.url;
+    const useExoPlayerPromise = this._getUseExoPlayer();
+    const masterPlaylistPromise = await fetch(url);
 
-      if (!hlsSupported) {
-        this._videoEl.innerHTML = this.hass.localize(
-          "ui.components.media-browser.video_not_supported"
-        );
-        return;
-      }
+    const hls = ((await import(
+      /* webpackChunkName: "hls.js" */ "hls.js"
+    )) as any).default as HLSModule;
+    let hlsSupported = hls.isSupported();
+
+    if (!hlsSupported) {
+      hlsSupported =
+        videoEl.canPlayType("application/vnd.apple.mpegurl") !== "";
     }
 
-    const url = this.url;
+    if (!hlsSupported) {
+      this._videoEl.innerHTML = this.hass.localize(
+        "ui.components.media-browser.video_not_supported"
+      );
+      return;
+    }
 
-    if (this._useExoPlayer) {
+    this._useExoPlayer = await useExoPlayerPromise;
+    const hevcRegexp = /CODECS=".*?((hev1)|(hvc1))\..*?"/;
+    const masterPlaylist = await masterPlaylistPromise.text();
+    if (hevcRegexp.test(masterPlaylist) && this._useExoPlayer) {
```

Comment:
Shouldnt we do this before loading and checking HLS?

Reply:
We could, and doing so would avoid the unnecessary loading of the HLS module in the case that we end up using ExoPlayer. But moving it up to before the HLS module load would likely be slower for the average case (I assume most cases will still use the hls.js polyfill or native) since we would block while waiting for the url to be fetched. Let me know which way you prefer.

Reply:
OK fair enough :-)

Reply:
Whoops, think it is blocking as is. The fetch() function involves network while the text() function should be fast right? Let me move the first await down.